File: ecommerce_api/src/main/java/com/example/ecommerce_api/enums/PaymentType.java
package com.example.ecommerce_api.enums;

import io.hms.api.exception.InvalidEnumValueException;

public enum PaymentType {

  Advance("Advance","Advance"),
  Paid("Paid","Paid"),
  Return("Return","Return");

  private final String key;
  private final String value;

  PaymentType(String key, String value) {
    this.key = key;
    this.value = value;
  }

  public String getKey() {
    return key;
  }
  public String getValue() {
    return value;
  }

  public static PaymentType getEnumByValue(String value) {
    for (PaymentType item : PaymentType.values()) {
      if (item.getValue().equals(value)) {
        return item;
      }
    }
    throw new InvalidEnumValueException();
  }
}

File: admin_portal/src/app/enum/PaymentType.ts
export class PaymentType{

  readonly key: string;
  readonly value: string;

  constructor(key: string, value: string) {
      this.key = key;
      this.value = value;
  }

  public static Advance = new PaymentType("Advance", "Advance");
  public static Paid = new PaymentType("Paid", "Paid");
  public static Return = new PaymentType("Return", "Return");
 
  static getAllEnumList(): Array<PaymentType> {
    return [
      PaymentType.Advance,
      PaymentType.Paid,
      PaymentType.Return,
    ];
  }
}